lib/pure/asyncftpclient.nim

```diff
@@ -293,17 +299,20 @@ proc retrFile*(ftp: AsyncFtpClient, file, dest: string,
   ## The ``EvRetr`` event is passed to the specified ``handleEvent`` function
   ## when the download is finished. The event's ``filename`` field will be equal
   ## to ``file``.
-  var destFile = open(dest, mode = fmWrite)
   await ftp.pasv()
-  var reply = await ftp.send("RETR " & file.normalizePathSep)
-  assertReply reply, ["125", "150"]
-  if {'(', ')'} notin reply.string:
-    raise newException(ReplyError, "Reply has no file size.")
+  var reply = await ftp.send("size " & file.normalizePathSep)
```

Comment:
More info here http://stackoverflow.com/questions/22151827/how-to-determine-the-end-of-a-file-when-using-retr-on-ftp-server-pasv